test/testchain/chain_builder.go

```diff
@@ -0,0 +1,87 @@
+// Copyright (c) 2024 The VeChainThor developers
+
+// Distributed under the GNU Lesser General Public License v3.0 software license, see the accompanying
+// file LICENSE or <https://www.gnu.org/licenses/lgpl-3.0.html>
+
+package testchain
+
+import (
+	"github.com/vechain/thor/v2/bft"
+	"github.com/vechain/thor/v2/chain"
+	"github.com/vechain/thor/v2/cmd/thor/solo"
+	"github.com/vechain/thor/v2/genesis"
+	"github.com/vechain/thor/v2/muxdb"
+	"github.com/vechain/thor/v2/state"
+)
+
+// ChainBuilder is a builder pattern struct used to construct a Chain instance.
+// It allows customization of the database, genesis configuration, and the consensus engine (BFT).
+type ChainBuilder struct {
+	dbFunc      func() *muxdb.MuxDB                        // Function to initialize the database.
+	genesisFunc func() *genesis.Genesis                    // Function to initialize the genesis configuration.
+	engineFunc  func(repo *chain.Repository) bft.Committer // Function to initialize the BFT consensus engine.
+}
+
+// NewIntegrationTestChain is a convenience function that creates a Chain for testing.
+// It uses an in-memory database, development network genesis, and a solo BFT engine.
+func NewIntegrationTestChain() (*Chain, error) {
```

Comment:
This one is good, but the ChainBuilder feels strange to me, kind of anti-pattern.wdyt?

Reply:
Is this still applicable @libotony ? I'm not following where the ChainBuilder is.

Reply:
I feel `ChainBuilder` is not necessary, personally more into `testchain.New(db, genesis, engine) *testchain.Chain`

Reply:
Gotcha ! Good catch, the ChainBuilder made sense in the first iteration of the feature. No longer makes sense 👍